internal/resources/client.go

```diff
@@ -0,0 +1,53 @@
+package resources
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"ldcli/internal/errors"
+)
+
+type Client interface {
```

Comment:
made a new interface for testing but had an issue with passing in data to the mock client, will try to tackle in follow up PR